src/llfs/storage_file_builder.test.cpp

```diff
@@ -56,6 +56,105 @@ class StorageFileBuilderTest : public ::testing::Test
+
+    // Iterate through each StorageFileConfigBlock
+    //
+    for (std::unique_ptr<llfs::StorageFileConfigBlock>& config_block : config_blocks) {
+      const llfs::PackedConfigBlock& p_config_block = config_block->get_const();
+      i64 expected_page0_offset =
+          llfs::PackedConfigBlock::kPayloadCapacity + sizeof(llfs::PackedConfigBlock::crc64);
+
+      // Iterate through each PackedPageDeviceConfig
+      //
+      for (const llfs::PackedConfigSlot& slot : p_config_block.slots) {
+        // Verify PackedPageDeviceConfig
+        //
+        BATT_ASSERT_EQ(slot.tag, llfs::PackedConfigSlotBase::Tag::kPageDevice);
+        const llfs::PackedPageDeviceConfig& page_device_config =
+            llfs::config_slot_cast<llfs::PackedPageDeviceConfig>(slot);
+        if (!(page_device_config.page_0_offset == expected_page0_offset &&
+              page_device_config.device_id == expected_device_id &&
+              page_device_config.page_count == kTestPageCount &&
+              page_device_config.page_size_log2 == page_size_log2)) {
+          return false;
+        }
+        // Update page offset numbers and device ID
+        //
+        expected_page0_offset +=
+            page_device_config.page_size() * kTestPageCount - llfs::PackedConfigSlot::kSize;
+        expected_device_id += 1;
+      }
+
+      // Set expected slots filled in config block. Full config block contains 62 slots
+      //
+      if (expected_device_id % 62 == 0) {
+        expected_num_slots = 62;
+      } else {
+        expected_num_slots = expected_device_id % 62;
+      }
+
+      // Set offsets of config block based on position
+      //
+      if (config_block == config_blocks.front()) {
+        expected_prev_offset = llfs::PackedConfigBlock::kNullFileOffset;
+      } else {
+        expected_prev_offset = -expected_next_offset;
+      }
+      if (config_block == config_blocks.back()) {
+        expected_next_offset = llfs::PackedConfigBlock::kNullFileOffset;
+      } else {
+        expected_next_offset =
+            batt::round_up_bits(12, expected_page0_offset + 64 * expected_num_slots);
+      }
+
+      // Verify the packed congig block
+      //
+      if (!(p_config_block.magic == llfs::PackedConfigBlock::kMagic &&
+            p_config_block.version == llfs::make_version_u64(0, 1, 0) &&
+            p_config_block.prev_offset == expected_prev_offset &&
+            p_config_block.next_offset == expected_next_offset &&
+            p_config_block.slots.size() == expected_num_slots &&
+            p_config_block.crc64 == p_config_block.true_crc64())) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void write_rand_data(llfs::IoRingRawBlockFile& test_file, i64 write_offset, i64 rand_buffer_len)
+  {
+    constexpr i64 kBufferSize = 512;
+    i64 rand_data_len = batt::round_up_bits(9, rand_buffer_len);
+    i64 offset_aligned = batt::round_up_bits(12, write_offset);
+    int rand_fd = open("/dev/random", O_RDONLY);
```

Comment:
Please, always check the return values of system calls for error.  Search for `status_from_retval` in the code to see examples of the right way to do this.